sentry-core/src/performance.rs

```diff
@@ -594,6 +702,163 @@ fn parse_sentry_trace(header: &str) -> Option<SentryTrace> {
+            None => return,
+        };
+
+        let mut name = shlib.name().to_string_lossy().to_string();
+        if name.is_empty() {
+            name = std::env::current_exe()
+                .map(|x| x.display().to_string())
+                .unwrap_or_else(|_| "<main>".to_string());
+        }
+
+        let code_id = shlib.id().map(|id| CodeId::new(format!("{}", id)));
+        let debug_name = shlib.debug_name().map(|n| n.to_string_lossy().to_string());
+
+        // For windows, the `virtual_memory_bias` actually returns the real
+        // `module_base`, which is the address that sentry uses for symbolication.
+        // Going via the segments means that the `image_addr` would be offset in
+        // a way that symbolication yields wrong results.
+        let (image_addr, image_vmaddr) = if cfg!(windows) {
+            (shlib.virtual_memory_bias().0.into(), 0.into())
+        } else {
+            (
+                shlib.actual_load_addr().0.into(),
+                shlib.stated_load_addr().0.into(),
+            )
+        };
+
+        images.push(
+            SymbolicDebugImage {
+                id: debug_id,
+                name,
+                arch: None,
+                image_addr,
+                image_size: shlib.len() as u64,
+                image_vmaddr,
+                code_id,
+                debug_file: debug_name,
+            }
+            .into(),
+        );
+    });
+
+    images
+}
+
+#[cfg(all(feature = "profiling", not(target_os = "windows")))]
+fn get_profile_from_report(
+    rep: &pprof::UnresolvedReport,
+    trace_id: TraceId,
+    transaction_id: sentry_types::Uuid,
+    transaction_name: String,
+) -> Profile {
+    let mut samples: Vec<Sample> = Vec::new();
+
+    for sample in rep.data.keys() {
+        let mut frames: Vec<RustFrame> = Vec::new();
+        for frame in &sample.frames {
+            frames.push(RustFrame {
+                instruction_addr: format!("{:p}", frame.ip()),
+            })
+        }
+        samples.push(Sample {
+            frames,
+            thread_name: String::from_utf8_lossy(&sample.thread_name[0..sample.thread_name_length])
+                .into_owned(),
+            thread_id: sample.thread_id,
+            nanos_relative_to_start: sample
+                .sample_timestamp
+                .duration_since(rep.timing.start_time)
+                .unwrap()
+                .as_nanos() as u64,
+        });
+    }
+    let sampled_profile = SampledProfile {
+        start_time_nanos: rep
+            .timing
+            .start_time
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_nanos() as u64,
+        start_time_secs: rep
+            .timing
+            .start_time
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_secs(),
+        duration_nanos: rep.timing.duration.as_nanos() as u64,
+        samples,
+    };
+    use std::time::SystemTime;
```

Comment:
interesting, I didn’t know you could put the `use` after the code that actually uses this.

Reply:
Right. This is a bit confusing. I'll move it to the top of the function `body`.